File: SOC/src/cc/game/soc/core/RouteType.java
package cc.game.soc.core;

public enum RouteType implements ILocalized {
	OPEN("Open", false, false, false),
    ROAD("Road", true, false, true),
	DAMAGED_ROAD("Damaged Road", true, false, true),
	SHIP("Ship", true, true, false),
	WARSHIP("Warship", true, true, false),
	;
	
	RouteType(String stringId, boolean isRoute, boolean isVessel, boolean isRoad) {
	    this.stringId  = stringId;
		this.isRoute = isRoute;
		this.isVessel = isVessel;
		this.isRoad = isRoad;
	}
	final String stringId;
	public final boolean isRoute;  // can this edge be counted as a route
	public final boolean isVessel; // is this edge a water vessel
	public final boolean isRoad;   // is the edge a road or damaged road


    @Override
    public String getName() {
        return stringId;
    }
}

File: SOC/src/cc/game/soc/core/RouteType.kt
package cc.game.soc.core

enum class RouteType(val _nameId: String, // can this edge be counted as a route
                     val isRoute: Boolean, // is this edge a water vessel
                     val isVessel: Boolean, // is the edge a road or damaged road
                     val isRoad: Boolean) : ILocalized {
	OPEN("Open", false, false, false),
	ROAD("Road", true, false, true),
	DAMAGED_ROAD("Damaged Road", true, false, true),
	SHIP("Ship", true, true, false),
	WARSHIP("Warship", true, true, false);

	override fun getNameId(): String = _nameId
}